src/services/config.ts

```diff
@@ -53,8 +53,9 @@ class Config {
   }
 
   getLocalConfig() {
+    const basePath = process.env.BASE_URL != '/' ? process.env.BASE_URL : '.'
```

Comment:
I cannot understand why you use logic like this. It looks too complicated and like you try to preserve the old behaviour with `.get('./config.json')`. But why we should preserve it? The old logic does not work properly and we must use the hack with the location of `config.json`. So I see two options here:
1. leave the old logic and continue using the Nginx hack.
2. use `BASE_URL` in the following manner: `const basePath = process.env.BASE_URL || '.'` (or something like this)

Reply:
The difficulty is that I have to consider that the web UI is deployed in many different environments, not just one identical to yours. My test environments alone include Heroku, AWS Cloudfront, GitHub pages and running it locally for development. The "old behaviour" works in all these environments and the majority of others.

It is only in environments where a base path other than the default is set that it the hack is required. And the default is '/' not null. See [publicPath config reference](https://cli.vuejs.org/config/#publicpath)

Reply:
As I know the hack is required in all environments and it does not depend on a base path because it fixes reloading the detail page. Reloading does not work because Alerta uses the relative URL to load `config.json` so on the details page the URL looks like `/alert/config.json` instead just `/config.json`. So your PR does not fix this problem. The PR fixes the problem only when we use a custom base path, to be precise.

So I suggest to not merge this change that solves the problem just partially and leave all as is because most of the people should continue using the hack anyway.

Reply:
And I cannot understand why you want to use a relative path here instead of an absolute one that is stored in the `BASE_URL` variable.